pkg/bootstrap/platform/platform.go

```diff
@@ -30,6 +32,13 @@ type Environment interface {
 	// Locality returns the run location for the bootstrap transformed from the
 	// platform-specific representation into the Envoy Locality schema.
 	Locality() *core.Locality
+
+	// Labels returns a collection of labels that exist on the underlying
+	// instance, structured as a map for label name to values.
+	Labels(meta *model.NodeMetadata) map[string]string
```

Comment:
can we do both of these without requiring NodeMetadata ?  It seems to only be used to get `PlatformMetadata`, which is provided by the `Metadata()` call on this same interface? 

I'd prefer to keep these interfaces (`Labels()` and `IsKubernetes()`) as clean as possible. It might mean we need to cache metadata for each environment, but I think any extra wiring inside the implementation would be worth it to make this more readable.

Reply:
Fixed